lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -542,7 +542,12 @@ function dosomething_campaign_get_campaigns_doing($uid = NULL) {
 function dosomething_campaign_get_campaign_block_vars($nid, $image_size = '740x480', $source = NULL) {
   $node = node_load($nid);
   $clc = dosomething_helpers_get_current_language_content_code();
-  $path_alias = drupal_get_path_alias('node/' . $nid);
+
+  global $user;
+  $language = dosomething_global_get_language($user, $node);
```

Comment:
thats the point of the function, it first checks if a translation exists in a users lang and then defers to the default language (global) and if nothing there then the source language 

https://github.com/DoSomething/phoenix/blob/dev/lib/modules/dosomething/dosomething_global/dosomething_global.module#L494